pkgs/development/tools/omnisharp-roslyn/default.nix

```diff
@@ -1,31 +1,70 @@
-{ lib, fetchFromGitHub, buildDotnetModule, dotnetCorePackages }:
-
+{ buildDotnetModule
+, dotnetCorePackages
+, fetchFromGitHub
+, icu
+, lib
+, patchelf
+, stdenv
+}:
 let
-  sdkVersion = dotnetCorePackages.sdk_6_0.version;
+  inherit (dotnetCorePackages) sdk_6_0;
 in
 buildDotnetModule rec {
   pname = "omnisharp-roslyn";
-  version = "1.38.2";
+  version = "1.39.1";
 
   src = fetchFromGitHub {
     owner = "OmniSharp";
     repo = pname;
     rev = "v${version}";
-    sha256 = "7XJIdotfffu8xo+S6xlc1zcK3oY9QIg1CJhCNJh5co0=";
+    sha256 = "Fd9fS5iSEynZfRwZexDlVndE/zSZdUdugR0VgXXAdmI=";
   };
 
   projectFile = "src/OmniSharp.Stdio.Driver/OmniSharp.Stdio.Driver.csproj";
   nugetDeps = ./deps.nix;
 
+  nativeBuildInputs = [
+    patchelf
+  ];
+
   dotnetInstallFlags = [ "--framework net6.0" ];
+  dotnetBuildFlags = [ "--framework net6.0" ];
+  dotnetFlags = [
+    # These flags are set by the cake build.
+    "-property:PackageVersion=${version}"
+    "-property:AssemblyVersion=${version}.0"
+    "-property:FileVersion=${version}.0"
+    "-property:InformationalVersion=${version}"
+    "-property:RuntimeFrameworkVersion=6.0.0-preview.7.21317.1"
```

Comment:
Can we get this from sdk_6_0=

Reply:
> Can we get this from sdk_6_0=

This is hardcoded in the omnisharp-roslyn build, it isn't fetched from the environment at all:
https://github.com/OmniSharp/omnisharp-roslyn/blob/850d92b06c461746e047dd09c3612622a7733090/build.cake#L268

So I believe fetching it from `sdk_6_0` would be a mistake (and the correct version would need to be fetched from `runtime_6_0` actually, which is what I was doing initially but ran into some issues with that)